tutorials/cloud-spanner-aspnet/index.md

```diff
@@ -0,0 +1,107 @@

This tutorial describes how to use our new C# Cloud Spanner ADO.NET library to build a simple REST API for Books.  We will use the asynchronous API of Cloud Spanner to store and query books. The Cloud Spanner ADO.NET library is fully asynchronous down to the network level to provide the best performance.
+
+What we’ll use in this blog post:
+
+* [Visual Studio 2017](https://www.visualstudio.com/vs/community/).  You can download the community edition for free from Microsoft.
+
+* [Cloud Spanner nuget package](https://github.com/GoogleCloudPlatform/google-cloud-dotnet).  You’ll add this to your project.
+
+### **Creating your Spanner Instance**
+
+In order to use Google Cloud Spanner, you'll first need to create a cloud project and download the Cloud SDK. The information on doing that can be found [here](https://cloud.google.com/spanner/docs/getting-started/set-up).
+
+Once you have a project,
+
+* Visit [https://](https://console.cloud.google.com/spanner/instances?project=)[console.cloud](https://console.cloud.google.com/spanner/instances?project=)[.google.com/spanner/instances?project=](https://console.cloud.google.com/spanner/instances?project=).
+
+* Create a new Spanner instance called "myspanner"
+
+* Create a database called "books" and then a new table called “bookTable”
+
+* Define the schema as below and set the primary key to "ID"
+
+![image alt text](image_0.png)
+
+### **Set up Default Application Credentials**
+
+### You need proper credentials to call Cloud Spanner, otherwise it will not know who is making the request. This sample uses Google Default Application Credentials to access your data.  Follow the steps online [here](https://developers.google.com/identity/protocols/application-default-credentials#howtheywork) to set up application credentials on your development machine.
+
+## **Cloud Spanner with ASP.NET**
+
+The first project we will create is an ASP.NET Core WebAPI application.  This REST API will provide basic CRUD operations on the bookTable created above. After we are done, you will be able to create, read, update and delete books.
+
+Create a new ASP.NET project via the menu **File**** ->** **New Project** -> **C#** -> **Web****->** **ASP.NET Core Web Application (.NET Core)**.
+
+Choose "Web API" and call the project “SpannerTest”.
+
+![image alt text](image_1.png)
+
+### **Add the nuget package for Spanner**
+
+The C# Spanner library for ADO.NET is available [on nuget] as "Google.Cloud.Spanner.Data". Add a reference to the Cloud Spanner nuget using the **Manage Nuget Packages** dialog.
+
+![image alt text](image_2.png)
+
+### **Model and Controller Class**
+
+The first code we need to write is a Model class for a Book.  Add a new folder called "Models" in the project and then a class called Book to that folder with the following content:
+
+    namespace SpannerTest.Models
```

Comment:
If we surround the code with
\```cs
\```
Do we get text coloring?

Reply:
Not yet.